drivers/nrf24l01/nrf24l01test.py

```diff
@@ -1,14 +1,40 @@
-"""Test for nrf24l01 module."""
-
-import struct
+"""Test for nrf24l01 module. Portable between MicroPython targets."""
+try:
+    import struct
+except ImportError:
+    import ustruct as struct
 import utime
 from machine import Pin, SPI
 from nrf24l01 import NRF24L01
+from sys import platform
+from micropython import const
+
+# Slave pause between receiving data and checking for further packets.
+_RX_POLL_DELAY = const(15)
+# Slave pauses an additional _SLAVE_SEND_DELAY ms after receiving data and before
+# transmitting to allow the (remote) master time to get into receive mode. The
+# master may be a slow device. Value tested with Pyboard, ESP32 and ESP8266.
+_SLAVE_SEND_DELAY = const(10)
+
+if platform == 'pyboard':
+    cfg = {'spi': 1, 'miso' : 'X7', 'mosi' : 'X8', 'sck' : 'X6', 'csn' : 'X5', 'ce' : 'X2'} # V2 PCB
+elif platform == 'esp8266':  # Hardware SPI
+    cfg = {'spi': 1, 'miso' : 12, 'mosi' : 13, 'sck' : 14, 'csn' : 4, 'ce' : 5}
+elif platform == 'esp32':  # Software SPI
+    cfg = {'spi': -1, 'miso' : 32, 'mosi' : 33, 'sck' : 25, 'csn' : 26, 'ce' : 27}
+else:
+    raise ValueError('Unsupported platform {}'.format(platform))
 
 pipes = (b'\xf0\xf0\xf0\xf0\xe1', b'\xf0\xf0\xf0\xf0\xd2')
 
 def master():
-    nrf = NRF24L01(SPI(2), Pin('Y5'), Pin('Y4'), payload_size=8)
+    csn = Pin(cfg['csn'], mode = Pin.OUT, value = 1)
+    ce = Pin(cfg['ce'], mode = Pin.OUT, value = 0)
+    if cfg['spi'] == -1:
+        spi = SPI(-1, sck=Pin(cfg['sck']), mosi=Pin(cfg['mosi']), miso=Pin(cfg['miso']))
+        nrf = NRF24L01(spi, csn, ce, payload_size=8, baudrate=100000)
```

Comment:
Does the baudrate need to be specified?  Won't soft SPI just use a maximum of 500kHz if the given value exceeds 500k?